test/unit-tests/components/viewmodels/roomlist/RoomListItemViewModel-test.tsx

```diff
@@ -138,6 +138,28 @@ describe("RoomListItemViewModel", () => {
         expect(vm.current.messagePreview).toBe(undefined);
     });
 
+    it("should check message preview when room change", async () => {
+        const otherRoom = mkStubRoom("roomId2", "roomName2", room.client);
+
+        jest.spyOn(MessagePreviewStore.instance, "getPreviewForRoom").mockResolvedValue({
+            text: "Message look like this",
+        } as MessagePreview);
+        mocked(useMessagePreviewToggle).mockReturnValue({
+            shouldShowMessagePreview: true,
+            toggleMessagePreview: jest.fn(),
+        });
+
+        const { result: vm, rerender } = renderHook((props) => useRoomListItemViewModel(props), {
+            initialProps: room,
+            ...withClientContextRenderOptions(room.client),
+        });
+        await waitFor(() => expect(vm.current.messagePreview).toBe(undefined));
```

Comment:
Shouldn't this be asserting that `vm.current.messagePreview` is `Message look like this`?